backend/app2/services/user_service.py

```
@@ -140,3 +142,78 @@ async def change_password(
+            # Update user profile with the new avatar URL
+            updated_profile = await self.update_profile(user_id, {"avatar_url": avatar_url})
+            logger.info(f"User profile updated with new avatar URL for user ID: {user_id}")
+            
+            return updated_profile # Return the full updated profile
+
+        except Exception as e:
+            logger.error(f"Failed to upload avatar or update profile for user {user_id}: {str(e)}", exc_info=True)
+            # Attempt to delete the file from storage if upload succeeded but DB update failed, to prevent orphans
+            # This part is tricky as we don't know at which stage the error occurred precisely without more checks.
+            # For simplicity, we are not adding cleanup logic here, but it's a consideration for robustness.
+            raise ServiceException(f"Failed to update avatar: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)